### DISCUSSION TITLE
Prompt Duplicated When Using Starship with Nushell

### DISCUSSION THREAD

**Discussion by KamilKleina** [nushell/nushell#16655] [Q&A]
### Question

Hi,

I'm encountering an issue where my prompt gets duplicated when using Starship with Nushell. I have the following line in my `config.nu`:

`use ($XDG_CONFIG_HOME + '/nushell/starship.nu')`

However, despite this configuration, the prompt is showing up twice.

Below are screenshots to illustrate the issue:

![image](https://github.com/user-attachments/assets/741a5133-a2ee-4fdb-8046-de155623e903)

Could anyone help me figure out how to resolve this and stop the prompt from being duplicated? I want use starship's prompt. 

If you need any more details or information from me, feel free to ask.

Thanks in advance!

BR,
Kamil

**Comment by fdncred** [comment]
I think you're talking about the prompt indicator. If you want it to be empty in nushell you need to set it like `$env.PROMPT_INDICATOR = {||""}`

**Comment by KamilKleina** [comment]
I don't want PROMPT_INDICATOR to be empty, I want to to understand why it's not delegated to starship 
I want prompt from starship to show if I am in vim normal/insert mode
I don't want to use promp from nushell

it does work correctly in zsh/bash but doesn't work correctly in nushell

**Comment by EdmundsEcho** [comment]
It is possible to hack a solution by disabling the nushell prompt.  However, to @KamilKleina 's point, nushell should properly delegate to starship.  This issue I'm running into is accessing the vi normal and insert modes.  In fish it reverses the starship prompt when in `normal` mode.  I can't get nushell to share the state of insert/normal mode with starship.  This is one example of the limitation.  

That all said, I don't know whether it is a starship or nushell issue.  Getting support for starship's vi mode indicator is a TODO in the `~/.cache/starship/init.nu` script.

**Comment by phucisstupid** [comment]
I have the same problem, hope this will be fix

**Comment by Kajiih** [comment]
Just referencing the related issue on Starship's repo, even if there isn't more progress: https://github.com/starship/starship/issues/5423

Currently, I think the best solution if we want to keep Starship's prompt (e.g. to use with multiple shells) is to do the following:

```nu
# config.nu

# Disable prompt from Nushell Because it is duplicated with that of Starship
$env.PROMPT_INDICATOR_VI_NORMAL = ""
$env.PROMPT_INDICATOR_VI_INSERT = ""

# Use cursor shapes to differentiate instead
$env.config.cursor_shape.vi_insert = "blink_line"
$env.config.cursor_shape.vi_normal = "blink_block"
```
